File: repo-root/groups/group-01-manage-field/code/managefield/src/main/java/com/jad/efreifive/managefield/dto/UpdateFieldStatusRequestDto.java
package com.jad.efreifive.managefield.dto;

import com.fasterxml.jackson.annotation.JsonProperty;
import jakarta.validation.constraints.NotBlank;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class UpdateFieldStatusRequestDto {

    @NotBlank(message = "status_id is required.")
    @JsonProperty("status_id")
    private String statusId;
}

File: repo-root/groups/group-01-manage-field/code/manage-field/src/main/kotlin/fr/efrei/managefield/controller/dto/request/UpdateFieldStatusRequestDto.kt
package fr.efrei.managefield.controller.dto.request

import com.fasterxml.jackson.annotation.JsonProperty
import jakarta.validation.constraints.NotBlank

/**
 * HTTP payload accepted when changing a field status.
 */
data class UpdateFieldStatusRequestDto(
    @field:NotBlank
    @JsonProperty("status_id")
    val statusId: String
)
